src/CodeGen_LLVM.cpp

```diff
@@ -2097,7 +2098,7 @@ void CodeGen_LLVM::visit(const Call *op) {
         vector<Value *> args = {user_context, char_ptr, codegen(op->args[1])};
 
         Value *buffer = codegen(op->args[2]);
-        buffer = builder->CreatePointerCast(buffer, buffer_t_type->getPointerTo());
+        buffer = builder->CreatePointerCast(buffer, debug_to_file->getFunctionType()->getParamType(3));
```

Comment:
I'm slightly dubious about this as it seems to be saying "cast the argument to whatever type the slot has" instead of casting to a type we know the item is. I don't fully understand the context so I may be wrong, but am concerned this may mask an error in the compiler if we are not careful.

Reply:
Under certain circumstances I was getting multiple copies of struct halide_buffer_t coming in under different names in the LLVM IR. Diagnosing seemed like a rabbit-hole, so I hit it with a hammer.